File: uikit/src/main/java/listeners/ComposeActionListener.java
package listeners;

import android.text.Editable;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;

import androidx.core.view.inputmethod.InputContentInfoCompat;

public abstract class ComposeActionListener {

    public void onMoreActionClicked(ImageView moreIcon) {}

    public void onPollActionClicked() {}

    public void onCameraActionClicked() {}

    public void onGalleryActionClicked() {}

    public void onAudioActionClicked() {}

    public void onFileActionClicked() {}

    public void onEmojiActionClicked(ImageView emojiIcon) {}

    public void onSendActionClicked(EditText editText ) {}

    public void onVoiceNoteComplete(String string) {}

    public abstract void  beforeTextChanged(CharSequence charSequence, int i, int i1, int i2);

    public abstract void onTextChanged(CharSequence charSequence, int i, int i1, int i2);

    public abstract void afterTextChanged(Editable editable);

    public void onEditTextMediaSelected(InputContentInfoCompat inputContentInfo) {};


    public void getCameraActionView(ImageView cameraIcon) {
        cameraIcon.setVisibility(View.VISIBLE);
    }

    public void getGalleryActionView(ImageView galleryIcon) {
        galleryIcon.setVisibility(View.VISIBLE);
    }

    public void getFileActionView(ImageView fileIcon) {
        fileIcon.setVisibility(View.VISIBLE);
    }

    public void getLocationActionView(ImageView locationIcon) {
        locationIcon.setVisibility(View.VISIBLE);
    }

    public void onLocationActionClicked() {
    }

    public void onStickerClicked() {
    }
}

File: uikit-kotlin/src/main/java/com/cometchat/pro/uikit/ui_components/shared/cometchatComposeBox/listener/ComposeActionListener.kt
package com.cometchat.pro.uikit.ui_components.shared.cometchatComposeBox.listener

import android.text.Editable
import android.view.View
import android.widget.EditText
import android.widget.ImageView
import androidx.core.view.inputmethod.InputContentInfoCompat

abstract class ComposeActionListener {
    fun onMoreActionClicked(moreIcon: ImageView?) {}
    open fun onCameraActionClicked() {}
    open fun onGalleryActionClicked() {}
    open fun onAudioActionClicked() {}
    open fun onFileActionClicked() {}
    open fun onLocationActionClicked() {}
    fun onEmojiActionClicked(emojiIcon: ImageView?) {}
    open fun onSendActionClicked(editText: EditText?) {}
    open fun onVoiceNoteComplete(string: String?) {}
    abstract fun beforeTextChanged(charSequence: CharSequence?, i: Int, i1: Int, i2: Int)
    abstract fun onTextChanged(charSequence: CharSequence?, i: Int, i1: Int, i2: Int)
    abstract fun afterTextChanged(editable: Editable?)

    open fun onEditTextMediaSelected(inputContentInfo: InputContentInfoCompat?) {}

    fun getCameraActionView(cameraIcon: ImageView) {
        cameraIcon.visibility = View.VISIBLE
    }

    fun getGalleryActionView(galleryIcon: ImageView) {
        galleryIcon.visibility = View.VISIBLE
    }

    fun getFileActionView(fileIcon: ImageView) {
        fileIcon.visibility = View.VISIBLE
    }

    open fun onStickerActionClicked() {}

    open fun onWhiteBoardClicked() {}
    open fun onWriteBoardClicked() {}
    open fun onStartCallClicked() {}
}